README.md

```diff
@@ -47,11 +113,199 @@ Parameters:
 --no-upgrade             Don't automatically run `occ upgrade` when the Updater finishes (note: `occ upgrade` is required after Updater updates the application code in order to push out any database changes in the newly deployed version of Nextcloud)
 ```
 
+## How it works
+
+Keeping Nextcloud Server up-to-date, at a high level, consists of three things:
+
+1. Updating the code
+2. Upgrading the database
+3. Updating all shipped and locally deployed apps
+
+The Updater app handles step #1 (see [*What does the updater do?*](https://docs.nextcloud.com/server/latest/admin_manual/maintenance/update.html#what-does-the-updater-do)) and prepares the environment for steps #2 and #3 (which are handled by `occ upgrade` *after* the Updater successfully completes).
+
+The Updater app is designed to be as independent from Server as possible for reliability purposes.
+
+### Theory of Operation
+
+* The Updater is wholly independent of the Server installation (save for being ran from `/updater/index.php` or `/updater/updater.phar`, which are
```

Comment:
```suggestion
* The Updater is wholly independent of the Server installation (safe for being ran from `/updater/index.php` or `/updater/updater.phar`, which are
```
This line is clearer, so I would remove the one above about being «as independent from Server as possible»

Reply:
```suggestion
* The Updater runs from either `/updater/index.php` (Web mode) or `/updater/updater.phar` (Command line mode), which are
```